syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -288,33 +304,17 @@ def mul_and_div(self, other, cmd):
             # 2) overflow when scaling self in division
 
             # sgn_self is 1 when new_self is positive else it's 0
-            # The comparison is different is new_self is a torch tensor or an AST
-            sgn_self = (
-                (new_self < self.field // 2).long()
-                if isinstance(new_self, torch.Tensor)
-                else new_self > 0
-            )
+            # The comparison is different if new_self is a torch tensor or an AST
+            sgn_self = (new_self > 0).type(self.torch_dtype)
             pos_self = new_self * sgn_self
-            neg_self = (
-                (self.field - new_self) * (1 - sgn_self)
-                if isinstance(new_self, torch.Tensor)
-                else new_self * (sgn_self - 1)
-            )
+            neg_self = new_self * (sgn_self - 1)
             new_self = neg_self + pos_self
 
             # sgn_other is 1 when new_other is positive else it's 0
-            # The comparison is different is new_other is a torch tensor or an AST
-            sgn_other = (
-                (new_other < self.field // 2).long()
-                if isinstance(new_other, torch.Tensor)
-                else new_other > 0
-            )
+            # The comparison is different if new_other is a torch tensor or an AST
+            sgn_other = (new_other > 0).type(self.torch_dtype)
             pos_other = new_other * sgn_other
-            neg_other = (
-                (self.field - new_other) * (1 - sgn_other)
-                if isinstance(new_other, torch.Tensor)
-                else new_other * (sgn_other - 1)
-            )
+            neg_other = new_other * (sgn_other - 1)
```

Comment:
I left it there initially because we were using sign related info below in some cases. 
But now upon reconsideration it is clear that it is just redundant with signed shares and should be removed to save computation.

Reply:
Update: Removing this is causing shape mismatch assertion error when linalg calls securenn.division. Need to investigate further. Keeping this logic in place for now.

Reply:
> shape mismatch

That's intriguing!